src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -109,8 +109,6 @@ private HttpURLConnection setupConnection() throws IOException {
             connection.setRequestProperty(header, mRequestHeaders.get(header));
         }
 
-        // Avoid reuse of existing sockets to avoid random EOF errors
-        System.setProperty("http.keepAlive", "false");
```

Comment:
I would strongly advice to remove this ASAP from the ADAL library. I hope there is an understanding of impact on consumers of a library by writing this code. In short this performed a global disable of the connection pipe-lining for every consumer of the ADAL library. For instance when client app downloads thumbnails it make new socket for each of them. Even if there is an EOF exception (I strongly it would be resolved by proposed fix), it still doesn't justify this code being written in the shared library.